### PR TITLE
Api integration

### PR DESCRIPTION
**Pull request by OlliPaj**
As requested

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,7 +33,7 @@
 		"publish-to-npm": "npm install && npm test && npm publish --access=public"
 	},
 	"dependencies": {
-		"@natlibfi/melinda-record-import-commons": "^1.3.0",
+		"@natlibfi/melinda-record-import-commons": "^3.1.1",
```
Comment (ghost) [inline package.json]: Update to the newest version (API authentication has changed and the new version of API client handles this).

File: .gitignore [hunk 2]
```diff
@@ -7,3 +7,5 @@ env
 .poll-change-timestamp.json
 fetched*
 src/helmet.js
+src/config.js
```
Comment (ghost) [inline .gitignore]: `src/config.js` should be included in version control (Configuration variables are from environment variables where all the sensitive data are). See [Helmet harvester](https://github.com/NatLibFi/melinda-record-import-harvester-helmet/blob/dev/src/config.js) for refence.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by OlliPaj** [comment]
Same style&version changes requested earlier in transformer.

File: .drone.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+name: default
+steps:
+- name: generate-tags
+  image: quay.io/natlibfi/drone-gen-tags
+- name: test-init
+  image: busybox
+  commands:
+  - chmod o+w /drone/src
+- name: test
+  image: node:8
```
Comment (ghost) [inline .drone.yml]: Change to `node:10`

File: src/harvest.js [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+		pollChangeTime = pollChangeTime || getPollChangeTime();
+
+		const timeBeforeFetching = moment();
+
+		logger.log('debug', `Fetching records updated between ${pollChangeTime.format()} - ${timeBeforeFetching.format()}`);
+		try {
+			await harvest(null);
+		} catch (e) {
+			fs.writeFileSync(failedHarvestFile, e); // If failure happens on n:th cycle (resumption) previous cycles (n-1) are already sent. Save data of harvest to file just in case used later.
+			throw (e);
```
Comment (ghost) [inline src/harvest.js]: Should be `throw e`

File: src/harvest.js [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+				timestamp: time.format()
+			}));
+		}
+
+		// ListRecords can only fetch 100 records at the time.
+		// Each cycle fetches max 100 records, filters records, passes them to callback and passes possible resumption token to new cycle
+		async function harvest(token) {
+			const url = new URL(harvestURL);
+
+			if (token) {
```
Comment (ghost) [inline src/harvest.js:95]: `URL`-instances have `searchParams`-property which can be used for setting the params (See [Helmet harvester](https://github.com/NatLibFi/melinda-record-import-harvester-helmet/blob/a7d8b6d30f92a7c8582e07edb2398e035502c0d3/src/harvest.js#L110) for reference).

File: src/harvest.js [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+				url.search = new URLSearchParams({
+					verb: 'ListRecords',
+					from: getPollChangeTime().utc().format(),
+					metadataPrefix: harvestMetadata
+				});
+				originalUrl = url.toString();
+			}
+
+			let response = null;
+			try {
```
Comment (ghost) [inline src/harvest.js]: The application should exit on request failure.

File: src/harvest.js [hunk 7]
```diff
@@ -0,0 +1,213 @@
...
+				if (failedQueries >= 5) {
+					throw new Error(JSON.stringify({time: moment(), query: url.toString(), queryOriginal: originalUrl, originalError: e}, null, 2));
+				}
+
+				return harvest(token || '');
+			}
+
+			failedQueries = 0;
+
+			if (response.status === HttpStatusCodes.OK) {
```
Comment (ghost) [inline src/harvest.js]: Split the block into functions so the code becomes easier to read.

Also `404` is not acceptable and should throw an error.

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/harvest.js (hunk 7 above)
Comment (OlliPaj) [reply src/harvest.js]: This is copied from Helmet, 404 else if block there too; https://github.com/NatLibFi/melinda-record-import-harvester-helmet/blob/master/src/harvest.js line 140

**Review by ghost — APPROVED** [review]

**Pull request merged**